usecases/collect_nodes_statuses.py

```diff
@@ -9,7 +9,7 @@
 class CollectNodesStatuses:
     """ Usecase class to collect nodes statuses
     """
-    def __init__(self, node_gateway: Union[NodeGateway, None] = None) -> None:
+    def __init__(self, node_gateway: Optional[NodeGateway] = None) -> None:
         self.node_gateway = node_gateway or NodeGateway()
 
     async def collect(self) -> None:
```

Comment:
As far as I understand, this method will trigger a call to the `node_data_aggregator` Lambda for each node, right?

Here Giovane told me that this happens in parallel: https://github.com/HathorNetwork/hathor-explorer-service/pull/3#pullrequestreview-673429229

And now we have added the aggregation logic in this same Lambda. So the aggregation logic will also be called for each node, and we will have parallel executions, if Giovane is right about parallelism.

Can't this lead to some nodes being missing from the final aggregated network? When comparing the old and new networks, you could be actually comparing one that was just saved by the other parallel executions of this Lambda?

Did you test this with several nodes to make sure this kind of problem doesn't happen?

Reply:
My understanding was that this should not happen asynchronously, since all aggregations are not atomic, without lock and write to the same key on redis. I will test this further to check which steps are truly asynchronous.

Reply:
Correction, this will all happen in parallel and it is possible that some execution may skip a node update, but since the nodes are saved on their own redis keys the next execution will aggregate with the status on the node it missed on the last aggregation.
So, although some errors may occur, it will correct itself in a few seconds.

If we want to prevent it from happening at all we could invoke the save lambda in sync and when all node statuses are collected (and saved) we invoke the aggregate lambda.

Reply:
This seems good enough for me for now